internal/logger/logrotate.go

```diff
@@ -86,40 +92,106 @@ func (w *Writer) Close() error {
 	return nil
 }
 
+var stdErrEnc = json.NewEncoder(os.Stderr)
+
 func (w *Writer) listen() {
 	for {
 		var r io.Reader = w.pr
 		if w.opts.MaximumFileSize > 0 {
 			r = io.LimitReader(w.pr, w.opts.MaximumFileSize)
 		}
 		if _, err := io.Copy(w.f, r); err != nil {
-			fmt.Println("Failed to write to log file", err)
+			msg := fmt.Sprintf("unable to write to log file %v: %w", w.f.Name(), err)
```

Comment:
```suggestion
			msg := fmt.Sprintf("unable to write to log file %v: %v", w.f.Name(), err)
```
`%w` doesn't work well outside of `fmt.Errorf`. 
> From Go playground
> ./prog.go:9:14: fmt.Sprintf does not support error-wrapping directive %w

Reply:
Oh yes it was renamed from fmt.Errorf()